spikeextractors/extractors/neoextractors/axonaextractor.py

```diff
@@ -0,0 +1,37 @@
+from .neobaseextractor import NeoBaseRecordingExtractor
+
+try:
+    import neo
+
+    HAVE_NEO = True
+except ImportError:
+    HAVE_NEO = False
+
+
+class AxonaRecordingExtractor(NeoBaseRecordingExtractor):
+    extractor_name = 'AxonaRecording'
+    mode = 'file'
+    NeoRawIOClass = 'AxonaRawIO'
+
+    def __init__(self, **kargs):
+        NeoBaseRecordingExtractor.__init__(self, **kargs)
+
+        # Read channel groups by tetrode IDs
+        self._key_properties['group'] = self.neo_reader.\
+            raw_annotations['blocks'][0]['segments'][0]['signals'][0][
+                '__array_annotations__']['tetrode_id']
```

Comment:
```suggestion
        self.set_channel_groups(groups=[x - 1 for x in self.neo_reader.raw_annotations['blocks'][0]['segments'][0]['signals'][0]['__array_annotations__']['tetrode_id']])
```
As a matter of practice, direct modification of internal attributes should really be avoided at all costs unless there is no other way to do what you're trying to do. Specific to channel groups, use the `set_channel_group` method for modifying the `key_properties['group']` field. For general properties, use `set_channel_property`.

Note the `-1` here also removes the need for the later lines.

Reply:
Good point!